tests/infra_test.go

```diff
@@ -678,7 +680,7 @@ var _ = Describe("[Serial]Infrastructure", func() {
 				Transport: tr,
 			}
 
-			errors := make(chan error)
+			errorsChan := make(chan error)
 			for ix := 0; ix < concurrency; ix++ {
 				go func(ix int) {
```

Comment:
This goroutine is unsafe as it may leak. The suggested change now will surely cause it to leak in case one of the errors are not of a timeout.

Reply:
can you elaborate on that? what leak are we talking about?

Reply:
i didnt touch it, how my changes / this PR scope related to it?

the for loop which i did change, runs  `concurrency` times, each one will be either an nil err , expected timeout or unexpected timeout.

please explain better what you mean and if its not belong to the scope of this PR, lets take it to other PR please.

Reply:
The changes here caused the receiver side of this groutine to change, so it is in the scope even if you did not touch these specific lines.

The channel is a blocking one (it has no buffers), therefore when `errorsChan <- err` is executed it will block until the receiving side will accept it.
But now, there receiver side will assert and may exit the loop **before** all responses have been processed. This will cause the goroutine to block, forever (i.e. leaking).

goroutines are dangerous, so if you touch it (including channels) make sure it is safe.

Reply:
there is a garbage collector that will close the channel when the context is done, and then the go routine will die sending on closed channel, but i will take care of it.

Reply:
> there is a garbage collector that will close the channel when the context is done, and then the go routine will die sending on closed channel, but i will take care of it.

If this is true why would you take care of it?
I am pretty sure the gc will not clean this. The gc does not close anything, it frees memory objects but does not call destruction. But if you find it is true, please share the ref.

Reply:
again deleted it before you answered because gc might not delete it since the channel is still used in the go routine,
btw we can buffer the channel as one of the solution or consume them all, implemented consuming atm.